src/psyclone/parse/module_manager.py

```diff
@@ -222,21 +243,145 @@ def ignores(self):
+        """
+
+        for fileinfo in self._filepath_to_file_info.values():
+            fileinfo: FileInfo
+            fileinfo.get_fparser_tree(verbose=verbose)
+
+    def load_all_psyir_nodes(self, verbose: bool = False) -> None:
+        """
+        Routine to load the psyir nodes of all files added
+        to the module manager
+
+        :param verbose: If `True`, print verbose information
+        """
+
+        for fileinfo in self._filepath_to_file_info.values():
+            fileinfo: FileInfo
+            fileinfo.get_psyir(verbose=verbose)
+
+    def load_all_module_infos(self, verbose: bool = False, indent: str = ""):
+        """Load the module info using psyir nodes
+
+        :param verbose: If `True`, print verbose information
+        :param indent: Prefix used as indentation for each line of
+            verbose output.
+
+        :raises KeyError: If module was already processed
+        """
+
+        # iterate over all file infos and load psyir
+        for file_info in self._filepath_to_file_info.values():
+            file_info: FileInfo
+
+            if verbose:
+                print(
+                    f"{indent}- Loading module information for "
+                    f"file '{file_info._filename}"
+                )
+
+            psyir_node: Node = file_info.get_psyir(
+                verbose=verbose, indent=indent + "  "
+            )
+
+            # Collect all module infos in this list
+            module_info_in_file: List[ModuleInfo] = []
+
+            # Walk over containers and add respective module information
+            for container_node in psyir_node.walk(
+                Container, stop_type=Routine
+            ):
+                if type(container_node) is not Container:
+                    # Sort out types which are not exactly of
+                    # type 'Container', e.g., 'FileContainer'
+                    continue
+
+                container_node: Container
+
+                container_name: str = container_node.name.lower()
+
+                if container_name in self._modules.keys():
+                    raise KeyError(
+                        f"Module '{container_name}' already processed"
+                    )
+
+                module_info = ModuleInfo(
+                    container_name, file_info, container_node
+                )
+                module_info_in_file.append(module_info)
+
+                self._modules[container_name] = module_info
+
+            filepath = file_info.filename
+            if filepath in self._filepath_to_module_info.keys():
+                raise KeyError(f"File '{filepath}' already processed")
+
+            self._filepath_to_module_info[filepath] = module_info_in_file
+
+    def get_all_module_infos(self) -> List[ModuleInfo]:
+        """
+        Return a list of all module infos
+        """
+
+        return list(self._modules.values())
+
+    def get_all_file_infos(self) -> List[FileInfo]:
```

Comment:
Am I right in thinking this doesn't do anything other than return information on objects we've already created? If so, I think this should be an `@property` named `all_file_infos`.